ext/hal/nxp/imx/drivers/gpio_imx.c

```diff
@@ -68,28 +70,33 @@ void GPIO_Init(GPIO_Type* base, const gpio_init_config_t* initConfig)
     else
     {
         icr = &GPIO_ICR2_REG(base);
-        pin -= 16;
+        icrShift -= 16;
     }
     switch(initConfig->interruptMode)
     {
         case(gpioIntLowLevel):
         {
-            *icr &= ~(0x3<<(2*pin));
+            *icr &= ~(0x3<<(2*icrShift));
             break;
         }
         case(gpioIntHighLevel):
         {
-            *icr = (*icr & (~(0x3<<(2*pin)))) | (0x1<<(2*pin));
+            *icr = (*icr & (~(0x3<<(2*icrShift)))) | (0x1<<(2*icrShift));
             break;
         }
         case(gpioIntRisingEdge):
         {
-            *icr = (*icr & (~(0x3<<(2*pin)))) | (0x2<<(2*pin));
+            *icr = (*icr & (~(0x3<<(2*icrShift)))) | (0x2<<(2*icrShift));
             break;
         }
         case(gpioIntFallingEdge):
         {
-            *icr |= (0x3<<(2*pin));
+            *icr |= (0x3<<(2*icrShift));
+            break;
+        }
+        case(gpioIntRisingOrFallingEdge):
+        {
+            GPIO_EDGE_SEL_REG(base) |= (1U << pin);
```

Comment:
There is a GPIO_SetIntEdgeSelect() function which does this setup.
It is possible to use gpioNoIntmode and call GPIO_SetIntEdgeSelect() afterwards, in which case the modification of this external BSP driver would perhaps be unnecessary.
But adding the new enum value gpioIntRisingOrFallingEdge seems cleaner.

Reply:
I don't think that using gpioNoIntmode reflects what we want to configure. This can be confusing for someone reading the code. I had to change this code anyway for the IMR and EDGE_SEL registers cleanup and the pin variable assignment.
I'll change it to call the GPIO_SetIntEdgeSelect() function instead.